extensions/blocks/eventbrite/edit.js

```diff
@@ -0,0 +1,397 @@
+import './editor.scss';
+
+const MODAL_BUTTON_STYLES = [
+	{ name: 'fill', label: __( 'Fill' ), isDefault: true },
+	{ name: 'outline', label: __( 'Outline' ) },
+];
+
+class EventbriteEdit extends Component {
+	state = {
+		editedUrl: this.props.attributes.url || '',
+		editingUrl: false,
+		// If this is a customized URL, we're going to need to find where it redirects to.
+		resolvingUrl: CUSTOM_URL_REGEX.test( this.props.attributes.url ),
+		resolvedStatusCode: null,
+	};
+
+	componentDidMount() {
+		const { resolvingUrl } = this.state;
+
+		// Check if we need to resolve an Eventbrite URL immediately.
+		if ( resolvingUrl ) {
+			this.resolveUrl();
+		}
+	}
+
+	componentDidUpdate( prevProps, prevState ) {
+		// Check if an Eventbrite URL has been entered, so we need to resolve it.
+		if ( ! prevState.resolvingUrl && this.state.resolvingUrl ) {
+			this.resolveUrl();
+		}
+	}
+
+	// TODO: figure out how to cancel request since apiFetch was updated to use Promises rather than XHR requests.
+	// componentWillUnmount() {}
+
+	resolveUrl = () => {
+		const { url } = this.props.attributes;
+
+		this.setState( { resolvedStatusCode: null } );
+
+		this.fetchRequest = apiFetch( {
+			path: `/wpcom/v2/resolve-redirect/${ url }`,
+		} );
+
+		this.fetchRequest.then(
+			response => {
+				// resolve
+				this.fetchRequest = null;
+				const resolvedUrl = response.url || url;
+				const resolvedStatusCode = response.status ? parseInt( response.status, 10 ) : null;
+
+				this.props.setAttributes( {
+					eventId: eventIdFromUrl( resolvedUrl ),
+					url: resolvedUrl,
+				} );
+				this.setState( {
+					resolvingUrl: false,
+					resolvedStatusCode,
+					editedUrl: resolvedUrl,
+				} );
+			},
+			xhr => {
+				// reject
+				if ( xhr.statusText === 'abort' ) {
+					return;
+				}
+				this.fetchRequest = null;
+				this.setState( {
+					resolvingUrl: false,
+					editingUrl: true,
+				} );
+			}
+		);
+	};
+
+	setUrl = event => {
+		if ( event ) {
+			event.preventDefault();
+		}
+
+		const { editedUrl: url } = this.state;
+
+		if ( ! url ) {
+			return;
+		}
+
+		this.props.setAttributes( {
+			eventId: eventIdFromUrl( url ),
+			url,
+		} );
+
+		// Setting the `resolvingUrl` state here, then waiting for `componentDidUpdate()` to
+		// be called before actually resolving it ensures that the `editedUrl` state has also been
+		// updated before resolveUrl() is called.
+		this.setState( {
+			editingUrl: false,
+			resolvingUrl: true,
+		} );
+	};
+
+	cannotEmbed = () => {
+		const { url } = this.props.attributes;
+		const { resolvedStatusCode } = this.state;
+
+		return (
+			( url && ! URL_REGEX.test( url ) ) || ( resolvedStatusCode && resolvedStatusCode >= 400 )
+		);
+	};
+
+	setEmbedType = embedType => {
+		const { setAttributes } = this.props;
+
+		setAttributes( { useModal: 'modal' === embedType } );
+	};
+
+	renderLoading() {
+		return (
+			<div className="wp-block-embed is-loading">
+				<Spinner />
+				<p>{ __( 'Embedding…' ) }</p>
```

Comment:
This can be translated as well.

Reply:
Tracked in https://github.com/Automattic/jetpack/issues/14458